app/models/proxy.rb

```diff
@@ -536,6 +536,9 @@ class PoliciesConfig
     include ActiveModel::Validations
     include Enumerable
 
+    # The policy chain must fit inside +ProxyConfig.content+, which is limited to 2.megabytes
+    MAX_LENGTH = 2.megabytes
```

Comment:
Hmm... tricky one. `ProxyConfig.content`consists not only of the policy chain, but also of other stuff. So, definitely PoliciesConfig's `MAX_LENGTH` must be less than content's max length, but it may happen that policy config is saved successfully, but then the proxy config cannot be generated out of it, because its size will be too big... 

But on the other hand - we can't predict the size of the "other" content (except the policy chain), so maybe that's OK.
Not sure though why 2MB was chosed as a limit, while the schema allows for 16MB.

But well, the good thing is that this limit is programmatic (not DB), so if in future it is not enough, we can raise it without performing another DB migration.

Reply:
About the 2MB limit, I just blindly obeyed @akostadinov.

Regarding the total content size limit. It's possible that the proxy config promotion fails because the total size of it it's too big. But as long as the error message says so, I don't think that's wrong. An admin facing such error can decide where to cut from, being it policies, metrics or anything else.

Reply:
audots would also be limitted to 16MB. And they contain both - old and new version of value. IIRC we discussed with Joan, he saw some other limit already at 2MB. Seems like a reasonable safe value that is high enough for existing customers, safe for 3scale components and as you said, easy to extend if needed.